### PR TITLE
remove duplicate child creation

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 blocktree_processor constructing grandchildren banks for every slot

 #### Summary of Changes
 remove the grandchild-ing code

### CODE REVIEW

File: core/src/blocktree_processor.rs [hunk 1]
```diff
@@ -193,11 +187,11 @@ pub fn process_blocktree(
                 bank_id: slot,
                 entry_height: starting_entry_height,
             };
-            fork_info.push((starting_bank, bfi));
+            fork_info.push((bank, bfi));
```
Comment (mvines) [inline core/src/blocktree_processor.rs]: This doesn't seem to unwind the entries from the partial slot.  I thought I wrote a test that would go red if we tried this

Comment (mvines) [reply core/src/blocktree_processor.rs]: Ah, this test would have probably failed with this line:
https://github.com/solana-labs/solana/blob/6cf6a1ccc3f05786692b5dab7ce0b98d5a65adbb/tests/multinode.rs#L48
But it's deleted now so problem solved

Comment (rob-solana) [reply core/src/blocktree_processor.rs]: roger, will fix

Comment (rob-solana) [reply core/src/blocktree_processor.rs]: this test test_process_blocktree_with_incomplete_slot should have covered it

still looking

File: core/src/blocktree_processor.rs [hunk 2]
```diff
@@ -212,16 +190,40 @@ pub fn process_blocktree(
...
+            // only process full slots in blocktree_processor, replay_stage
+            //  handles any partials
+            if next_meta.is_full() {
+                let next_bank = Arc::new(Bank::new_from_parent(
+                    &bank,
+                    leader_schedule_utils::slot_leader_at(next_slot, &bank),
+                    next_slot,
+                ));
+                trace!("Add child bank for slot={}", next_slot);
+                // bank_forks.insert(*next_slot, child_bank);
```
Comment (mvines) [inline core/src/blocktree_processor.rs]: nit: remove commented out code

File: core/src/blocktree_processor.rs [hunk 3]
```diff
@@ -116,30 +116,24 @@ pub fn process_blocktree(
...
-            starting_bank.slot(),
-        ));
-        let mut entry_height = starting_entry_height;
 
         // Load the metadata for this slot
         let meta = blocktree
             .meta(slot)
             .map_err(|err| {
-                warn!("Failed to load meta for slot {}: {:?}", slot, err);
+                eprintln!("Failed to load meta for slot {}: {:?}", slot, err);
```
Comment (mvines) [inline core/src/blocktree_processor.rs]: why `eprintln`?

Comment (mvines) [reply core/src/blocktree_processor.rs]: ack, our println banner is looking in `src/`, which no longer exists . 📉

**Pull request merged**